specs/clipboard-api/clipboard.idl

```diff
@@ -0,0 +1,426 @@
+  [marshaling_behavior(standard)]
+  static runtimeclass Clipboard
+  {
+    static DataPackageView GetContent();
+
+    static void SetContent(DataPackage content);
+    static Boolean SetContentWithOptions(DataPackage content, ClipboardContentOptions options);
+    static void Flush();
+    static void Clear();
+
+    static event Windows.Foundation.EventHandler<Object> ContentChanged;
+
+    static Windows.Foundation.IAsyncOperation<ClipboardHistoryItemsResult> GetHistoryItemsAsync();
+    static SetHistoryItemAsContentStatus SetHistoryItemAsContent(ClipboardHistoryItem item);
+    static Boolean DeleteItemFromHistory(ClipboardHistoryItem item);
+    static Boolean ClearHistory();
+
+    static Boolean IsHistoryEnabled();
+    static Boolean IsRoamingEnabled();
+
+    static event Windows.Foundation.EventHandler<Object> HistoryChanged;
+    static event Windows.Foundation.EventHandler<Object> RoamingEnabledChanged;
+    static event Windows.Foundation.EventHandler<Object> HistoryEnabledChanged;
+  }
+
+  /// <summary>
+  /// Clipboard history-related settings your app can set for a new item
+  /// that it cuts or copies onto the clipboard using
+  /// <see cref="Clipboard.SetContentWithOptions"/>.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass ClipboardContentOptions
+  {
+    ClipboardContentOptions();
+
+    Boolean IsRoamable;
+    Boolean IsAllowedInHistory;
+    Windows.Foundation.Collections.IVector<String> RoamingFormats { get; };
+    Windows.Foundation.Collections.IVector<String> HistoryFormats { get; };
+  }
+
+  /// <summary>
+  /// An item retrieved from the clipboard history log.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass ClipboardHistoryItem
+  {
+    String Id { get; };
+    Windows.Foundation.DateTime Timestamp { get; };
+    DataPackageView Content { get; };
+  }
+
+  /// <summary>
+  /// Return value for a call to <see cref="Clipboard.GetHistoryItemsAsync"/>.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass ClipboardHistoryItemsResult
+  {
+    ClipboardHistoryItemsResultStatus Status { get; };
+    Windows.Foundation.Collections.IVectorView<ClipboardHistoryItem> Items { get; };
+  }
+
+  /// <summary>
+  /// Basic unit of data transfer for <see cref="Clipboard"/>.
+  /// Allows data to be sent in multiple "formats". Certain formats can be
+  /// "delay rendered", meaning they are not serialized immediately into
+  /// the DataPackage, but instead a creator-provided callback will
+  /// serialize them on demand.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass DataPackage
+  {
+    DataPackage();
+
+    DataPackageView GetView();
+    DataPackagePropertySet Properties { get; };
+    DataPackageOperation RequestedOperation;
+
+    event Windows.Foundation.TypedEventHandler<DataPackage, OperationCompletedEventArgs> OperationCompleted;
+    event Windows.Foundation.TypedEventHandler<DataPackage, Object> Destroyed;

```

Comment:
We should probably delete this event. The [documentation for DataPackage.Destroyed](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.datatransfer.datapackage.destroyed?view=winrt-19041) says "In general, the system handles the destruction of a DataPackage object. Your app should not have to handle this event." I agree as I can't imagine any use for it not already covered by OperationCompleted. Plus it confuses the strange ownership model of DataPackage (does the system own it? does the source app?) even more.

But before deleting the event, I should dig up the original Windows 8-era internal API specs to see why the event was added originally.